mavros/include/mavros/plugin.hpp

```diff
@@ -74,18 +74,9 @@ class Plugin : public std::enable_shared_from_this<Plugin>
   //! Subscriptions vector
   using Subscriptions = std::vector<HandlerInfo>;
 
-  explicit Plugin(UASPtr uas_)
-  : uas(uas_), node(std::dynamic_pointer_cast<rclcpp::Node>(uas_))
-  {}
-
   explicit Plugin(
-    UASPtr uas_, const std::string & subnode,
-    const rclcpp::NodeOptions & options = rclcpp::NodeOptions())
-  : uas(uas_),
-    // node(std::dynamic_pointer_cast<rclcpp::Node>(uas_)->create_sub_node(subnode))  // https://github.com/ros2/rclcpp/issues/731
-    node(rclcpp::Node::make_shared(subnode,
-      std::dynamic_pointer_cast<rclcpp::Node>(uas_)->get_fully_qualified_name(), options))
-  {}
+    UASPtr uas, const std::string & subnode_name,
+    const rclcpp::NodeOptions & options = rclcpp::NodeOptions());
```

Comment:
Why to move that into cpp?

Reply:
S.t. all plugins don't have to recompile when something changes here.

Reply:
Why? I don't see any advantage of having it in the header.